src/legacy/core_plugins/kibana/public/dashboard/np_ready/dashboard_empty_screen_constants.tsx

```diff
@@ -19,6 +19,13 @@
 
 import { i18n } from '@kbn/i18n';
 
+/** READONLY VIEW CONSTANTS **/
+export const emptyDashboardTitle: string = i18n.translate('kbn.dashboard.emptyDashboardTitle', {
+  defaultMessage: 'This dashboard is empty or you do not have permission to view.',
```

Comment:
Don't we know concretely that it's both, not one of these? Would suggest breaking this up to be able to provide direct messaging (You probably only need "This dashboard is empty" and "You do not have permission to view this dashboard"; "This dashboard is empty and you don't have permission to see it" seems a bit overkill.)

I think direct, unambiguous error/help messages are always appreciated even if it requires a few more conditionals on our side.